gradle-plugins/biz.aQute.bnd.gradle/src/main/java/aQute/bnd/gradle/BundleTaskExtension.java

```diff
@@ -208,9 +210,13 @@ public BundleTaskExtension(org.gradle.api.tasks.bundling.Jar task) {
 		classpath = objects.fileCollection();
 		allSource = objects.fileCollection();
 		outputDirectory = objects.directoryProperty();
+
 		SourceSet mainSourceSet = sourceSets(project).getByName(SourceSet.MAIN_SOURCE_SET_NAME);
 		setSourceSet(mainSourceSet);
-		classpath(mainSourceSet.getCompileClasspath());
+
+		Configuration mainCompileClasspath = task.getProject().getConfigurations().getByName(mainSourceSet.getCompileClasspathConfigurationName());
+		requestJarLibrary(mainCompileClasspath, objects);
```

Comment:
This can work here because the task extension maintains its own classpath file collection to enable to the user to configure it differently than the main compile classpath.